shared/src/telemetry/data.rs

```diff
@@ -0,0 +1,55 @@
+use serde_json::json; // Ensure you have `serde_json` in your Cargo.toml
+use serde_json::Value;
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+
+// Function to get the PostgreSQL data directory from the PGDATA environment variable
+pub fn get_postgres_data_directory() -> Option<String> {
+    env::var("PGDATA").ok()
+}
+
+pub fn read_telemetry_data(extension_name: String) -> Result<Value, String> {
+    // Determine the base PostgreSQL data directory
+    let pg_data_directory =
+        get_postgres_data_directory().ok_or("PGDATA environment variable is not set")?;
+
+    // Customize the path based on the extension name
+    let directory_path = PathBuf::from(pg_data_directory).join(match extension_name.as_str() {
```

Comment:
Don't do this, just pass a `&Path` into this function as an argument

Reply:
I pondered doing both, but the problem is that the way we retrieve the path via `ParadeDirectory` and `WriterDirectory` is dependent on the struct being initialized. Since we call the background worker from `PG_init`, it is never initialized and doesn't exist. So I would need to do the `PGDATA` + path append in some module in each extension, then pass the path here, duplicating the code. I thought this was cleaner, but if you disagree I can change it